src/timely-util/src/builder_async.rs

```diff
@@ -301,14 +259,43 @@ where
     C: Container,
     P: Push<Bundle<T, C>> + 'static,
 {
-    #[allow(clippy::unused_async)]
-    pub async fn give<CP, D>(&mut self, cap: &CP, data: D)
+    pub fn give<D>(&mut self, cap: &Capability<T>, data: D)
     where
-        CP: CapabilityTrait<T>,
         CapacityContainerBuilder<C>: PushInto<D>,
     {
         let mut handle = self.handle.borrow_mut();
-        cap.session(&mut handle).push_into(data);
+        handle.session_with_builder(cap).give(data);
+    }
+}
+
+impl<T, D, P>
+    AsyncOutputHandle<T, AccountedStackBuilder<CapacityContainerBuilder<StackWrapper<D>>>, P>
+where
+    D: timely::Data + Columnation,
+    T: Timestamp,
+    P: Push<Bundle<T, StackWrapper<D>>>,
+{
+    pub const MAX_OUTSTANDING_BYTES: usize = 128 * 1024 * 1024;
+
+    /// Provides one record at the time specified by the capability. This method will automatically
+    /// yield back to timely after [Self::MAX_OUTSTANDING_BYTES] have been produced.
+    pub async fn give_fueled<D2>(&mut self, cap: &Capability<T>, data: D2)
```

Comment:
I believe this will always be considered for inlining (at least according to this article https://matklad.github.io/2021/07/09/inline-in-rust.html#Inlining-in-Rust) because the method is generic